llvm/lib/Transforms/Scalar/GVNSink.cpp

```diff
@@ -719,12 +719,11 @@ GVNSink::analyzeInstructionForSinking(LockstepReverseIterator &LRI,
   // try and continue making progress.
   Instruction *I0 = NewInsts[0];
 
-  // If all instructions that are going to participate don't have the same
-  // number of operands, we can't do any useful PHI analysis for all operands.
-  auto hasDifferentNumOperands = [&I0](Instruction *I) {
-    return I->getNumOperands() != I0->getNumOperands();
+  auto hasDifferentOperands = [&I0](Instruction *I) {
```

Comment:
nit: I'd rename this to something like...
```suggestion
  auto isNotSameOperation = [&I0](Instruction *I) {
```
...as the (non-type) operands may still be different.